### ISSUE TITLE
Cop Idea: "when" not "if"

### ISSUE THREAD

**Issue by mockdeep** [backus/rubocop-rspec#400]
I don't know if this is the sort of thing that would result in a lot of false positives, but I like conditions in a spec description to use consistent wording:

```ruby
# good
it 'does this when that'

# bad
it 'does this if that'

# good
describe 'when that' do
  it 'does this'

# bad
describe 'if that' do
  it 'does this'
```

**Comment by pirj** [comment]
`context` is a better fit for examples that need specific conditions.
`ContextWording` and some cop that would suggest extracting conditional part of the docstring into a context according to [Use Contexts](https://rspec.rubystyle.guide/#use-contexts) guideline would do the job in this case.

However, I foresee interference with [No context for single `it` block](https://github.com/rubocop-hq/rubocop-rspec/issues/178) cop.

**Comment by mockdeep** [comment]
Yeah, we generally don't use `context` if there's only a single test for a condition.
